lib/cassandra/types.rb

```diff
@@ -727,6 +740,16 @@ def to_s
         "map<#{@key_type.to_s}, #{@value_type.to_s}>"
       end
 
+      def hash
+        @hash ||= begin
+          h = 17
+          h = 31 * h + kind.hash
```

Comment:
With my proposal, all 'kind' references in the file are basically replaced with @kind. In this case, it would be better because if we ever changed the Map's kind to something else, we only change it in the constructor and don't have to worry about other references to :map.